packages/dashboard/src/customization/widgets/text/component.test.tsx

```diff
@@ -13,88 +11,101 @@ jest.mock('~/store/actions', () => ({
-      const { isSelected, readOnly, isUrl } = configuration;
-      (useIsSelected as jest.Mock).mockImplementation(() => isSelected);
-      (useSelector as jest.Mock).mockImplementation(() => readOnly);
-      (useDispatch as jest.Mock).mockReturnValue(jest.fn());
-
-      const { container } = render(
-        <TextWidgetComponent
-          id='some-id'
-          x={1}
-          y={2}
-          z={3}
-          height={100}
-          width={100}
-          type='text-widget'
-          properties={{ isUrl, value: 'abc' }}
-        />
-      );
-
-      expect(container).toMatchSnapshot();
-    });
+  it('should render editable text', () => {
+    const text = 'text';
+    render(<TestEditableTextWidget value={text} />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toBeVisible();
+    expect(textWidget).not.toHaveAttribute('aria-readonly');
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
   });
 
-  it('should exit edit mode when unmounted', () => {
-    (useDispatch as jest.Mock).mockImplementation(() => jest.fn((cb) => cb())); // short curcuit dispatch
-
-    (useIsSelected as jest.Mock).mockImplementation(() => false);
-    (useSelector as jest.Mock).mockImplementation(() => false);
-    (useDispatch as jest.Mock).mockReturnValue(jest.fn());
-
-    const { unmount } = render(
-      <TextWidgetComponent
-        id='some-id'
-        x={1}
-        y={2}
-        z={3}
-        height={100}
-        width={100}
-        type='text-widget'
-        properties={{ isUrl: false, value: 'abc' }}
-      />
-    );
-    unmount();
-
-    expect(onChangeDashboardGridEnabledAction).toBeCalledWith({
-      enabled: true,
-    });
+  it('should render editable link', () => {
+    const text = 'text';
+    const href = 'https://test.com';
+    render(<TestEditableTextWidget value={text} href={href} isUrl />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toHaveTextContent(text);
+    expect(textWidget).not.toHaveAttribute('aria-readonly');
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
+  });
+
+  it('should santize and then render unsafe link', () => {
+    const text = 'text';
+    const href = 'javascript://%0Aalert(1)';
+    render(<TestReadonlyTextWidget value={text} href={href} isUrl />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toBeVisible();
+    expect(textWidget).not.toHaveAttribute('href');
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
+  });
+
+  it('should santize and then render unsafe link 2', () => {
+    const text = 'text';
+    const href = '\x6A\x61\x76\x61\x73\x63\x72\x69\x70\x74\x3aalert(1)';
+    render(<TestReadonlyTextWidget value={text} href={href} isUrl />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toBeVisible();
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
```

Comment:
[Super nit]: This assertion is not in the same place as the other tests (i.e., 2nd expect vs last in the others). Very minor thing, but it sticks out as "different", when it is not.